### ISSUE TITLE
Modal does not contain Focus Cycling mechanism

### ISSUE THREAD

**Issue by jeffredodd** [SAP/fundamental-ngx#477]
### Description
In order to be WCAG 2.1 compliant, our Modal's with the `role='dialog'` must contain a mechanism to enforce focus cycling. A users focus must be contained within a modal to be compliant. See https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal for more information.  

### Reproduction
1) Open a modal
2) Attempt to tab

Expected:
- Focus should remain within modal
- Focus should be redirected into modal when opened

Actual:
- Focus is not altered and left in last known state. 

_Note: This issue is seen in all 3 libraries_
Fundamental-React: SAP/fundamental-react#345
Fundamental-NGX: SAP/fundamental-ngx#477
Fundamental-Vue: SAP/fundamental-vue#136
